### DISCUSSION TITLE
Boot partition is so small.

### DISCUSSION THREAD

**Discussion by imsakg** [lima-vm/lima#2429] [General]
Hey, I just looked at discussions and issue about boot partition size problem but I didn't found any. I like to learn why images default /boot partition size is 128mb and FAT16? Today's kernel can't fit in it and FAT16 thing is kinda old fashioned. I tried to grow /boot partition but it overlapping with /root partition. Can someone explain it and help me about it?

BTW, I tried to upgrade Arch Linux and that's when I faced the problem.

**Comment by afbjorklund** [comment]
I think this depends on the distribution, but I have also seen the issue.

The workaround that I ended up with, on a non-Lima installation, was to shrink the root partition and add a new and bigger UEFI boot partition at the end of it. This does mean that old partition is unused, and that it complains about them being in wrong order.


EDIT: The default instance (Ubuntu) does have a bigger /boot partition:

```
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1        96G  1.4G   95G   2% /
/dev/vda16      881M   61M  758M   8% /boot
/dev/vda15      105M  6.2M   99M   6% /boot/efi
```

**Reply by afbjorklund** [reply]
Arch Linux doesn't seem to have any boot partition (and legacy BIOS), so there was plenty of space.

```
[anders@lima-archlinux lima]$ df -h /boot
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda3       100G  1.4G   98G   2% /
```

**Comment by madalinignisca** [comment]
Lima does not do a preseeded install of any distribution. Like other virtualization tools, it takes the cloud image and creates a disk resizing the roofs only. This is how all public clouds do it, except that they might build their own images.

If you have any distribution that does not fit the default partitioning, and you think that it should, that needs to be discussed upstream with the selected provider.
